lms/djangoapps/lms_xblock/runtime.py

```diff
@@ -137,7 +138,11 @@ def __init__(self, **kwargs):
         store = modulestore()
 
         services = kwargs.setdefault('services', {})
-        services['completion'] = CompletionService(user=kwargs.get('user'), course_key=kwargs.get('course_id'))
+        user = kwargs.get('user')
+        course_key = kwargs.get('course_id')
+        services['completion'] = CompletionService(user=user, course_key=course_key)
+        if isinstance(user, AnonymousUser):
+            services['completion'] = None
```

Comment:
We shouldn't initialize the service if we don't need it. Is there a need to put a `None` in? Or is the following sufficient:

```
user = kwargs.get('user')
if user.is_authenticated:
  services['completion'] = CompletionService(user=kwargs.get('user'), course_key=kwargs.get('course_id'))
```

Reply:
I was just being extra cautious as I wasn't sure if that change might affect any other code using it :-).
I've changed it now. Thanks.